lib/debezium/types.go

```diff
@@ -95,57 +95,19 @@ func toInt64(value any) (int64, error) {
 	return 0, fmt.Errorf("failed to cast value '%v' with type '%T' to int64", value, value)
 }
 
-func (f Field) shouldCastToInt64() bool {
-	if f.IsInteger() {
-		return true
-	}
-
-	switch f.DebeziumType {
-	case
-		Timestamp,
-		MicroTimestamp,
-		Date,
-		Time,
-		MicroTime,
-		DateKafkaConnect,
-		TimeKafkaConnect,
-		DateTimeKafkaConnect:
-		// We have verfied that these [Field.DebeziumType]s are always used with a [Field.Type] of [Int16], [Int32], or [Int64].
-		return true
-	}
-
-	return false
-}
-
 func (f Field) ParseValue(value any) (any, error) {
 	if value == nil {
 		return nil, nil
 	}
 
 	// Preprocess [value] to reverse the effects of being JSON marshalled and unmarshalled when passing through Kafka.
-	if f.shouldCastToInt64() {
+	switch f.Type {
+	case Int16, Int32, Int64:
 		var err error
 		value, err = toInt64(value)
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		// TODO: If we don't see the following error in the logs replace [Field.shouldCastToInt64] call with a simple
-		// check that [Field.Type] is [Int16], [Int32], or [Int64].
-		switch f.Type {
-		case Int16, Int32, Int64:
-			slog.Error(fmt.Sprintf("Field type is %q but shouldCastToInt64 is false", f.Type), slog.Any("value", value))
-		}
-	}
-
-	// Check if the field is an integer and requires us to cast it as such.
-	if f.IsInteger() {
```

Comment:
Move this to https://github.com/artie-labs/transfer/pull/463/files#diff-6f8a144d4abeb92f8240e55383e15b8323e83bb3b9f69088274ed61f4f344e96R155

We want to handle `DebeziumType`s first before we handle `Type`.